src/app/components/Profile/Profile.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from 'react';
+import defaultAvatar from './defaultAvatar.svg';
+import classes from './Profile.module.css';
+import { Profile, getCurrentUser } from '../../utils/user';
+
+const defaultProfile: Profile = {
+  username: 'Game-Ticker',
+  displayName: 'Game-Ticker',
+  avatar: null,
+};
+
+function openLoginDialog() {
+  overwolf.profile.openLoginDialog();
+}
+
+const User = (): JSX.Element => {
+  const [profile, setProfile] = useState<Profile>(defaultProfile);
+  const [profileError, setProfileError] = useState<string | null>(null);
+  const [loginDialog, setLoginDialog] = useState(false);
+  const [login, setLogin] = useState(false);
+
+  useEffect(() => {
+    overwolf.profile.onLoginStateChanged.addListener((result) => {
```

Comment:
If this component is unmounted, the listener still exists.
It's a good practice to clean up:
https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect

```ts
() => {
  overwolf.profile.onLoginStateChanged.removeListener(handleLoginStateChanged);
}
```

Reply:
Yes, thats true, i wasnt sure how to do this correctly., I will reimplement this